crates/block-producer/src/block_builder/mod.rs

```diff
@@ -130,47 +255,79 @@ impl BlockBuilder {
+    block: Block,
+}
 
-        let new_block_header = self.block_kernel.prove(block_header_witness)?;
+impl SelectedBlock {
+    fn inject_telemetry(&self) {
+        let span = Span::current();
+        span.set_attribute("block.number", i64::from(self.block_number.as_u32()));
+        span.set_attribute("block.batches.count", i64::from(self.batches.len() as u32));
+    }
+}
 
-        // TODO: return an error?
-        let block =
-            Block::new(new_block_header, updated_accounts, output_notes, produced_nullifiers)
-                .expect("invalid block components");
+impl BlockSummaryAndInputs {
+    fn inject_telemetry(&self) {
+        let span = Span::current();
 
-        let block_hash = block.hash();
-        let block_num = new_block_header.block_num();
+        // SAFETY: We do not expect to have more than u32::MAX of any count per block.
+        span.set_attribute(
+            "block.updated_accounts.count",
+            i64::try_from(self.summary.updated_accounts.len())
+                .expect("less than u32::MAX account updates"),
+        );
+        span.set_attribute(
+            "block.output_notes.count",
+            i64::try_from(self.summary.output_notes.len())
+                .expect("less than u32::MAX output notes"),
+        );
+        span.set_attribute(
+            "block.nullifiers.count",
+            i64::try_from(self.summary.nullifiers.len()).expect("less than u32::MAX nullifiers"),
+        );
+        span.set_attribute(
+            "block.dangling_notes.count",
+            i64::try_from(self.summary.dangling_notes.len())
+                .expect("less than u32::MAX dangling notes"),
+        );
+    }
+}
 
-        info!(target: COMPONENT, %block_num, %block_hash, "block built");
-        debug!(target: COMPONENT, ?block);
+impl ProvenBlock {
+    fn inject_telemetry(&self) {
+        let span = Span::current();
+        let header = self.block.header();
 
-        self.store
-            .apply_block(&block)
-            .await
-            .map_err(BuildBlockError::StoreApplyBlockFailed)?;
+        span.set_attribute("block.hash", header.hash().to_hex());
+        span.set_attribute("block.sub_hash", header.sub_hash().to_hex());
+        span.set_attribute("block.parent_hash", header.prev_hash().to_hex());
 
-        info!(target: COMPONENT, %block_num, %block_hash, "block committed");
+        span.set_attribute("block.protocol.version", i64::from(header.version()));
 
-        Ok(())
+        span.set_attribute("block.roots.kernel", header.kernel_root().to_hex());
+        span.set_attribute("block.roots.nullifier", header.nullifier_root().to_hex());
+        span.set_attribute("block.roots.account", header.account_root().to_hex());
+        span.set_attribute("block.roots.chain", header.chain_root().to_hex());
+        span.set_attribute("block.roots.note", header.note_root().to_hex());
+        span.set_attribute("block.roots.transation", header.tx_hash().to_hex());
```

Comment:
```suggestion
        span.set_attribute("block.roots.transaction", header.tx_hash().to_hex());
```

Reply:
Thanks. I've also renamed `roots` to `commitment`.